arch/arm/include/cortex_m/tz.h

```diff
@@ -283,6 +283,9 @@ int tz_sau_region_configure(tz_sau_conf_t *p_sau_conf);
  */
 typedef void __attribute__((cmse_nonsecure_call)) (*tz_ns_func_ptr_t) (void);
 
+/* Required for C99 compilation */
```

Comment:
Does this not need an ifdef?

Reply:
I just reverted #16199
Apparently the ifdef was not present there

Reply:
Added the ifndef